otelhelpers/context_traceparent.go

```diff
@@ -0,0 +1,87 @@
+// otelhelpers is a package of helper functions for dealing with otel
+// traceparent propagation over files and environment variables.
+package otelhelpers
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+
+	"go.opentelemetry.io/otel"
+)
+
+// ContextWithEnvTraceparent is a helper that looks for the the TRACEPARENT
+// environment variable and if it's set, it grabs the traceparent and
+// adds it to the context it returns. When there is no envvar or it's
+// empty, the original context is returned unmodified.
+func ContextWithEnvTraceparent(ctx context.Context) context.Context {
+	traceparent := os.Getenv("TRACEPARENT")
+	if traceparent != "" {
+		return ContextWithTraceparentString(ctx, traceparent)
+	}
+	return ctx
+}
+
+// ContextWithLinuxCmdlineTraceparent looks in /proc/cmdline for a traceparent=
+// command line option and returns the context with that value as traceparent
+// if it's there. Does no validation. Returns the original context if there is
```

Comment:
Why not validate here? Callers need to handle the no-op case anyway. TODO rather than as part of this PR?

Reply:
Caller can validate all it wants, the idea here is to be "first do no harm" and to avoid having to pass back an error. Generally when an otel thing fails the best thing to do is let it slide so it doesn't disturb the non-tracing code.

Reply:
But in this case we could pass back an invalid traceparent instead of silently dropping it which might be worse?